fdisk: build fails on IA32 architectures

## fdisk fails for for all tested arch/cpu types emitting following error:

In file included from fdiskbsdlabel.c:62:0:
fdiskbsdlabel.h:61:2: error: #error unknown architecture
fdiskbsdlabel.c: In function 'bselect':
fdiskbsdlabel.c:180:44: error: 'BSD_LABELSECTOR' undeclared (first use in this function)
fdiskbsdlabel.c:180:44: note: each undeclared identifier is reported only once for each function it appears in
fdiskbsdlabel.c: In function 'xbsd_write_bootstrap':
fdiskbsdlabel.c:539:24: error: 'BSD_LABELSECTOR' undeclared (first use in this function)
fdiskbsdlabel.c: In function 'xbsd_readlabel':
fdiskbsdlabel.c:744:28: error: 'BSD_LABELSECTOR' undeclared (first use in this function)
fdiskbsdlabel.c:744:60: error: 'BSD_LABELOFFSET' undeclared (first use in this function)
fdiskbsdlabel.c: In function 'xbsd_writelabel':
fdiskbsdlabel.c:769:32: error: 'BSD_LABELSECTOR' undeclared (first use in this function)
fdiskbsdlabel.c:780:61: error: 'BSD_LABELOFFSET' undeclared (first use in this function)
## make[2]: **\* [fdisk-fdiskbsdlabel.o] Error 1

Tested architectures:
- i486 using gcc -march=i486
- i586 using gcc -march=pentium-mmx
- i686 using gcc -march={pentiumpro,pentium2,pentium3,pentium-m,pentium4}

Tested with following major toolchain components:

binutils 2.21.53.0.2
gcc 4.6.2
glibc 2.14.1

--- a/schedutils/chrt.c
+++ b/schedutils/chrt.c
@@ -527,7 +527,7 @@ int main(int argc, char **argv)
 	if (ctl->policy == SCHED_RR)
 		need_prio = true;
 
-	if (ctl->verbose)
+	if (ctl->pid > -1 && ctl->verbose)
 		show_sched_info(ctl);
 
 	bool have_prio = need_prio ||

--- a/schedutils/chrt.c
+++ b/schedutils/chrt.c
@@ -510,9 +510,6 @@ int main(int argc, char **argv)
 
 	/* If option --pid was given, parse the very last argument as a PID. */
 	if (ctl->pid == 0) {
-		if (need_prio && argc - optind < 2)
-			errx(EXIT_FAILURE, _("policy %s requires a priority argument"),
-						get_policy_name(ctl->policy));
 		errno = 0;
 		/* strtopid_or_err() is not suitable here, as 0 can be passed. */
 		ctl->pid = strtos32_or_err(argv[argc - 1], _("invalid PID argument"));
@@ -530,13 +527,17 @@ int main(int argc, char **argv)
 	if (ctl->pid > -1 && ctl->verbose)
 		show_sched_info(ctl);
 
-	bool have_prio = need_prio ||
-		(ctl->pid == -1 ? (optind < argc && isdigit_string(argv[optind])) : (argc - optind > 1));
-
-	if (have_prio) {
+	if (argc - optind > 1 && (ctl->pid > -1 || isdigit_string(argv[optind]))) {
 		errno = 0;
 		ctl->priority = strtos32_or_err(argv[optind], _("invalid priority argument"));
-	} else
+
+		optind++;
+	} else if (need_prio && ctl->pid == -1 && isdigit_string(argv[optind]))
+		errx(EXIT_FAILURE, _("no command or priority specified"));
+	else if (need_prio)
+		errx(EXIT_FAILURE, _("policy %s requires a priority argument"),
+					get_policy_name(ctl->policy));
+	else
 		ctl->priority = 0;
 
 	if (ctl->runtime && !supports_runtime_param(ctl->policy))
@@ -575,11 +576,6 @@ int main(int argc, char **argv)
 
 	if (!ctl->pid) {
 		argv += optind;
-
-		if (need_prio)
-			argv++;
-		else if (argv[0] && isdigit_string(argv[0]))
-			argv++;
 
 		if (argv[0] && strcmp(argv[0], "--") == 0)
 			argv++;

